File: CS/MyXtraGrid/Custom View/MyGridViewInfo.cs
using System;
using System.Drawing;
using DevExpress.XtraGrid.Views.Grid.ViewInfo;
using DevExpress.XtraGrid.Views.Grid;
using DevExpress.XtraGrid.Drawing;

namespace MyXtraGrid {
	public class MyGridViewInfo : GridViewInfo {
		public MyGridViewInfo(DevExpress.XtraGrid.Views.Grid.GridView gridView) : base(gridView) {}

        public new MyGridView View { get { return base.View as MyGridView; } }

        public override int CalcRowHeight(Graphics graphics, int rowHandle, int visibleIndex, int min, int level, bool useCache, GridColumnsInfo columns)
        {
            int height = base.CalcRowHeight(graphics, rowHandle, visibleIndex, min, level, useCache, columns);
            int paddingHeight = View.CellPadding.Vertical;
            return height + paddingHeight;
        }
        public override int CalcColumnBestWidth(DevExpress.XtraGrid.Columns.GridColumn column)
        {
            int width = base.CalcColumnBestWidth(column);
            int paddingWidth = View.CellPadding.Horizontal;
            return width + paddingWidth;
        }

        protected override GridCellInfo CalcRowCellDrawInfoCore(GridDataRowInfo ri, GridColumnInfoArgs ci, GridCellInfo cell, GridColumnInfoArgs nextColumn, bool calcEditInfo, GridRow nextRow, bool allowCache)
        {
            GridCellInfo result = base.CalcRowCellDrawInfoCore(ri, ci, cell, nextColumn, calcEditInfo, nextRow, allowCache);
            Rectangle rect = cell.CellValueRect;
            rect.X += View.CellPadding.Left;
            rect.Width -= View.CellPadding.Horizontal;
            rect.Y += View.CellPadding.Top;
            rect.Height -= View.CellPadding.Vertical;
            cell.CellValueRect = rect;
            return result;
        }
	}
}

File: VB/MyXtraGrid/Custom View/MyGridViewInfo.vb
Imports System
Imports System.Drawing
Imports DevExpress.XtraGrid.Views.Grid.ViewInfo
Imports DevExpress.XtraGrid.Views.Grid
Imports DevExpress.XtraGrid.Drawing

Namespace MyXtraGrid
	Public Class MyGridViewInfo
		Inherits GridViewInfo

		Public Sub New(ByVal gridView As DevExpress.XtraGrid.Views.Grid.GridView)
			MyBase.New(gridView)
		End Sub

		Public Shadows ReadOnly Property View() As MyGridView
			Get
				Return TryCast(MyBase.View, MyGridView)
			End Get
		End Property

		Public Overrides Function CalcRowHeight(ByVal graphics As Graphics, ByVal rowHandle As Integer, ByVal visibleIndex As Integer, ByVal min As Integer, ByVal level As Integer, ByVal useCache As Boolean, ByVal columns As GridColumnsInfo) As Integer
			Dim height As Integer = MyBase.CalcRowHeight(graphics, rowHandle, visibleIndex, min, level, useCache, columns)
			Dim paddingHeight As Integer = View.CellPadding.Vertical
			Return height + paddingHeight
		End Function
		Public Overrides Function CalcColumnBestWidth(ByVal column As DevExpress.XtraGrid.Columns.GridColumn) As Integer
			Dim width As Integer = MyBase.CalcColumnBestWidth(column)
			Dim paddingWidth As Integer = View.CellPadding.Horizontal
			Return width + paddingWidth
		End Function

		Protected Overrides Function CalcRowCellDrawInfoCore(ByVal ri As GridDataRowInfo, ByVal ci As GridColumnInfoArgs, ByVal cell As GridCellInfo, ByVal nextColumn As GridColumnInfoArgs, ByVal calcEditInfo As Boolean, ByVal nextRow As GridRow, ByVal allowCache As Boolean) As GridCellInfo
			Dim result As GridCellInfo = MyBase.CalcRowCellDrawInfoCore(ri, ci, cell, nextColumn, calcEditInfo, nextRow, allowCache)
			Dim rect As Rectangle = cell.CellValueRect
			rect.X += View.CellPadding.Left
			rect.Width -= View.CellPadding.Horizontal
			rect.Y += View.CellPadding.Top
			rect.Height -= View.CellPadding.Vertical
			cell.CellValueRect = rect
			Return result
		End Function
	End Class
End Namespace
